frontend/javascript/components/postupload/app-shell.vue

```diff
@@ -0,0 +1,340 @@
+  steps: Array
+})
+
+const progressDesktop = computed(() => props.step / (props.steps.length - 1))
+
+const progressMobile = computed(() => (props.step + 1) / props.steps.length)
+
+const onlineHelpShow = ref(false)
+const onlineHelpPath = ref('')
+
+const showhelp = (e) => {
+  onlineHelpShow.value = true
+  onlineHelpPath.value = e
+}
+</script>
+
+<template>
+  <div class="appshell">
+    <header class="appshell-header">
+      <div
+        class="appshell-header-container appshell-header-container--mobile container">
+        <div class="appshell-header-logo">
+          <div class="appshell-header-logo-image">
+            <span class="visually-hidden">FragDenStaat</span>
+          </div>
+        </div>
+        <div class="appshell-header-title1">
+          <slot name="mobile-header-title1" />
+        </div>
+        <div class="appshell-header-title2">
+          <slot name="mobile-header-title2" />
+        </div>
+      </div>
+      <div class="appshell-header-container appshell-header-container--desktop">
+        <div class="appshell-header-steps my-3">
+          <ol>
+            <li
+              v-for="(stepLabel, stepIndex) in steps"
+              :key="stepIndex"
+              class="appshell-header-step"
+              :class="{ 'appshell-header-step--active': stepIndex <= step }">
+              <div class="appshell-header-step-marker">{{ stepIndex + 1 }}</div>
+              <div>{{ stepLabel }}</div>
+            </li>
+          </ol>
+          <div
+            class="appshell-header-progress appshell-header-progress--desktop">
+            <div
+              class="appshell-header-progress-bar"
+              :style="{ width: progressDesktop * 100 + '%' }"></div>
+          </div>
+        </div>
+      </div>
+      <div class="appshell-header-progress appshell-header-progress--mobile">
+        <div
+          class="appshell-header-progress-bar"
+          :style="{ width: progressMobile * 100 + '%' }"></div>
+      </div>
+    </header>
+    <div class="appshell-nav container">
+      <slot name="nav" />
+    </div>
+    <div class="appshell-main">
+      <slot name="main" @showhelp="showhelp" />
+    </div>
+    <div class="appshell-help" v-if="onlineHelpShow">
+      <!-- .close should be provided by bootstrap, hmm -->
+      <button
+        type="button"
+        class="close appshell-help-close"
+        @click="onlineHelpShow = false"></button>
+      <online-help :path="onlineHelpPath" />
+    </div>
+    <div class="appshell-actions">
+      <slot name="actions" />
+    </div>
+  </div>
+</template>
+
+<!-- unscoped!
+  well, scoped to the current django route .../post-hochladen/ since the global css is not ejected+collected
+  else would have to guard via body.appshell
+-->
+<style lang="scss">
+// TODO: arbitrarily switch to "appshell" under this breakpoint
+//   not really usable, but also not broken. WIP
+// TODO: sync this with PostUpload's matchMedia and scss
+$breakpoint: 1000px;
```

Comment:
better use the bootstrap provided layout, grid system, utilities and variables for consistency and maintainability

Reply:
this one is complex :)
- appshell needs to hide `#header`, but header from the base template can't have class `d-lg-none` globally, only when appshell is used...
- by default, BS doesn't have responsive `position-*-fixed` classes. we could [enable them](https://getbootstrap.com/docs/5.3/utilities/api/#enable-responsive)
- but I do more in the `@media` blocks, like `top`, `bottom` (which could be worked around, I guess)
- before I found out about `position-foo-fixed` I implemented one global scss+css var in froide/frontend/styles/variables: `$froide-appshell-breakpoint` which equals BS's `lg` -- because IMO it makes sense to have it defined centrally/DRY. It wouldn't make sense to change *one* of the `d-lg-foo` in appshell; they need to be in sync. This adheres to BS's grid breakpoints without suggesting that individual `d-lg-foo` classes can be adjusted individually. (If we go with this solution I'll still have to fix the logic; I used `max-width` while BS's breakpoints all cascade `min-width`ly...)